File: guides/micronaut-data-json-duality-view/java/src/main/java/example/micronaut/domain/Student.java
package example.micronaut.domain;

import io.micronaut.core.annotation.Nullable;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;
import io.micronaut.data.annotation.sql.JoinTable;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Collections;
import java.util.List;

@MappedEntity(value = "TBL_STUDENT", alias = "s") // <1>
public record Student (
        @Id // <2>
        @GeneratedValue(GeneratedValue.Type.IDENTITY) // <3>
        Long id,
        String name,

        @JoinTable(name = "TBL_STUDENT_CLASSES", alias = "sc") // <4>
        @Relation(Relation.Kind.MANY_TO_MANY) // <5>
        List<Class> classes
) {}

File: guides/micronaut-data-json-duality-view/kotlin/src/main/kotlin/example/micronaut/domain/Student.kt
package example.micronaut.domain

import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import io.micronaut.data.annotation.sql.JoinTable

@MappedEntity(value = "TBL_STUDENT", alias = "s") // <1>
data class Student(
    @Id // <2>
    @GeneratedValue(GeneratedValue.Type.IDENTITY) // <3>
    val id: Long?,
    val name: String,
    @JoinTable(name = "TBL_STUDENT_CLASSES", alias = "sc") // <4>
    @Relation(Relation.Kind.MANY_TO_MANY) // <5>
    val classes: List<Class>,
    @field:JsonAnyGetter // <6>
    @field:JsonAnySetter
    val extras: Map<String, Any?> = emptyMap()
)
